src/ui/third_party/binsider.rs

```diff
@@ -0,0 +1,108 @@
+use crate::debugger::Debugger;
+use crate::ui::tui::app::port::DebuggerEventQueue;
+use crate::ui::tui::TuiHook;
+use crate::ui::{console, supervisor, DebugeeOutReader};
+use binsider::prelude::{Command, Event};
+use binsider::{prelude::*, tui::ui::Tab};
+use ratatui::{
+    crossterm::event::{self, Event as CrosstermEvent, KeyCode},
+    Frame,
+};
+use std::fs;
+use std::path::PathBuf;
+use std::sync::mpsc;
+use std::time::Duration;
+
+pub struct AppBuilder {
+    debugee_out: DebugeeOutReader,
+    debugee_err: DebugeeOutReader,
+}
+
+impl AppBuilder {
+    pub fn new(debugee_out: DebugeeOutReader, debugee_err: DebugeeOutReader) -> Self {
+        Self {
+            debugee_out,
+            debugee_err,
+        }
+    }
+
+    pub fn extend(self, mut debugger: Debugger) -> BinsiderApplication {
+        let debugger_event_queue = DebuggerEventQueue::default();
+        debugger.set_hook(TuiHook::new(debugger_event_queue.clone()));
+
+        BinsiderApplication::new(debugger, self.debugee_out, self.debugee_err)
+    }
+}
+
+/// This is a wrapper over `binsider` application (see https://github.com/orhun/binsider) for running
+/// it inside a `BugStalker`.
+pub struct BinsiderApplication {
+    debugger: Debugger,
+    debugee_out: DebugeeOutReader,
+    debugee_err: DebugeeOutReader,
+}
+
+impl BinsiderApplication {
+    fn new(
+        debugger: Debugger,
+        debugee_out: DebugeeOutReader,
+        debugee_err: DebugeeOutReader,
+    ) -> Self {
+        Self {
+            debugger,
+            debugee_out,
+            debugee_err,
+        }
+    }
+
+    pub fn run(self) -> anyhow::Result<supervisor::ControlFlow> {
+        let path = PathBuf::from(self.debugger.process().program());
+        let file_data = fs::read(self.debugger.process().program())?;
+        let file_info = FileInfo::new(
+            path.to_str().unwrap_or_default(),
+            None,
+            file_data.as_slice(),
+        )?;
+        let analyzer = Analyzer::new(file_info, 15, vec![])?;
+        let mut state = State::new(analyzer)?;
+        let (sender, receiver) = mpsc::channel();
+        state.analyzer.extract_strings(sender.clone());
+
+        let mut terminal = ratatui::init();
+        loop {
+            // Render the UI.
+            terminal.draw(|frame: &mut Frame| {
+                render(&mut state, frame);
+            })?;
+
+            // Handle terminal events.
+            if event::poll(Duration::from_millis(16))? {
+                if let CrosstermEvent::Key(key) = event::read()? {
+                    if key.code == KeyCode::Char('q') {
+                        break;
+                    }
+                    let command = Command::from(key);
+                    state.run_command(command, sender.clone())?;
```

Comment:
You can also disable some commands here FYI - e.g. I'm not sure selecting dynamic dependencies work as expected.

Reply:
Yep, I made it as simple as possible for two reasons:
1) this is a development branch of the `bs` and I think its ok to have some non-working functionality for now
2) i expect that we will be able to make all the functionality work before the next 0.x release of `bs` :)